File: src/main/java/coffee/amo/interactionkey/registry/KeyRegistry.java
package coffee.amo.interactionkey.registry;

import net.minecraft.client.KeyMapping;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.client.event.RegisterKeyMappingsEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import org.lwjgl.glfw.GLFW;

import static coffee.amo.interactionkey.InteractionKey.MODID;

@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD, modid = MODID, value = Dist.CLIENT)
public class KeyRegistry {
    public static final KeyMapping INTERACTION_KEY = new KeyMapping("key.interaction_key.interaction_key", GLFW.GLFW_KEY_F, "key.categories.interaction_key");

    @SubscribeEvent
    public static void onKeyMappingEvent(RegisterKeyMappingsEvent event){
        event.register(INTERACTION_KEY);
    }
}

File: src/main/kotlin/coffee/amo/interactionkey/registry/KeyRegistry.kt
package coffee.amo.interactionkey.registry

import coffee.amo.interactionkey.InteractionKey
import net.minecraft.client.KeyMapping
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.client.event.RegisterKeyMappingsEvent
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod.EventBusSubscriber
import org.lwjgl.glfw.GLFW

@EventBusSubscriber(bus = EventBusSubscriber.Bus.MOD, modid = InteractionKey.MODID, value = [Dist.CLIENT])
object KeyRegistry {
    @JvmField
    val INTERACTION_KEY = KeyMapping("key.interaction_key.interaction_key", GLFW.GLFW_KEY_F, "key.categories.interaction_key")
    val SCROLL_UP_KEY = KeyMapping("key.interaction_key.scroll_up_key", GLFW.GLFW_KEY_UP, "key.categories.interaction_key")
    val SCROLL_DOWN_KEY = KeyMapping("key.interaction_key.scroll_down_key", GLFW.GLFW_KEY_DOWN, "key.categories.interaction_key")
    @SubscribeEvent
    fun onKeyMappingEvent(event: RegisterKeyMappingsEvent) {
        event.register(INTERACTION_KEY)
        event.register(SCROLL_UP_KEY)
        event.register(SCROLL_DOWN_KEY)
    }
}